sdk/openai/inference/src/auth/azure_key_credential.rs

```diff
@@ -0,0 +1,62 @@
+use async_trait::async_trait;
+use std::sync::Arc;
+
+use azure_core::{
+    auth::Secret,
+    headers::{HeaderName, HeaderValue},
+    Context, Header, Policy, PolicyResult, Request,
+};
+
+/// A key credential for the [AzureOpenAIClient](crate::clients::AzureOpenAIClient).
+///
+/// # Example
+/// ```no_run
+/// use azure_openai_inference::clients::{AzureOpenAIClient, AzureOpenAIClientMethods};
+///
+/// let secret = std::env::var("OPENAI_KEY").expect("Set OPENAI_KEY env variable");
+/// let azure_open_ai_client = AzureOpenAIClient::with_key_credential(
+///    "https://my.endpoint/",
+///     secret,
+///     None,
+/// ).unwrap();
+/// ```
+#[derive(Debug, Clone)]
+pub struct AzureKeyCredential(Secret);
```

Comment:
This will go into `azure_core`, but we are renaming the `auth` module to `credentials`.